source/h2_frames.c

```diff
@@ -754,26 +761,47 @@ static int s_frame_prebuilt_encode(
     struct aws_byte_buf *output,
     bool *complete) {
 
+    (void)encoder;
     struct aws_h2_frame_prebuilt *frame = AWS_CONTAINER_OF(frame_base, struct aws_h2_frame_prebuilt, base);
 
-    if (frame->send_progress.len == frame->encoded_buf.len) {
+    if (frame->cursor.len == frame->encoded_buf.len) {
```

Comment:
```/* frame->cursor and frame->encoded_buf are initialized to the same storage buffer */```

Maybe some version of that makes this clearer? Also, I see that's how it works (I THINK), but then there's a frame prefix written right after initialization, so how do the cursor and encoded buf ever match again? My brain compiler is not advanced enough to work it out at a glance.

Reply:
I'll add more comments explaining it all